lib/SPIRV/libSPIRV/SPIRVEntry.cpp

```diff
@@ -270,13 +271,30 @@ void SPIRVEntry::addDecorate(Decoration Kind, SPIRVWord Literal) {
   addDecorate(new SPIRVDecorate(Kind, this, Literal));
 }
 
+void SPIRVEntry::addDecorateId(SPIRVDecorateId *Dec) {
```

Comment:
It's a duplication of "addDecorate" and their bodies can be merged actually. But the way I've done it now IMO is more transparent. Same is applicable to all Decorate functions below. I don't mind to change it tho.

Reply:
I'd prefer to have only `addDecorate()`

Reply:
.. or call one from another.